### ISSUE TITLE
Resource mapping file - bug?

### ISSUE THREAD

**Issue by jikoriko** [Azure/aztfy#114]
Hi,

I am currently working on creating a shell in Terraform of an APIM for a new environment / subscription based off one that was manually created and configured in another, when I run aztfy on the resource group there are more than 1000 resources as there are a lot of API's deployed through it ect. 

I have managed to filter this down in the resource mapping file to 22 resources we want Terraform to create apart of this shell, when I input that file to aztfy it still marks to import all the resources it could automatically map but I can see that a few resources I needed to manually map such as identity providers are now mapped. I though the resource mapping file would be explicit to the only resources aztfy would go to import, is this expected behavior?

For now I will work through the list removing the mapped entries for the resources we don't want to import, also as some possible enhancements from my use so far of aztfy which looks like a great tool in its early stages:

 1) DEL key : remove the generated mapping for a selected resource, i.e skipping it. This would make it a lot easier going through the list and marking resources to be skipped.
 2) Be able to filter on the mapped names, i.e filter down to resources that are mapped to a azurerm_resource_group.

**Comment by magodo** [comment]
@jikoriko Can you try the `main` branch? We recently consolidate the resource mapping file in https://github.com/Azure/aztfy/pull/107 to avoid automatically deduce TF resource types when users have provided the resource mapping file. Whilst, the resources exist in the resource group but is not specified in the resource mapping file will still show up in the import list, but marked as to be skipped.

For the proposal for the enhancement:
1. DEL key: that's a good idea! I'll keep this issue open for tracking this feature
2. Filter: CUrrently, you can filter the resource id via <kbd>/</kbd>, which accepts regexp, I think this is more useful than filter the mapped type?

**Comment by jikoriko** [comment]
Hi @magodo, thanks for getting back so quickly. 

I am currently using the aztfy_v0.4.0_windows_amd64 binary which was updated 3 days ago, this does line up on the same date of #107 so that change could be missing in that build? when I save the resource mapping file I am not seeing all skipped ones in that file with empty TF type strings, perhaps I should have a look at building it from the master from the recent changes you have made :)

On note of filtering regexp could help out, its not a huge issue not been able filter on TF resource type but I thought if the filtering could look at both that would be good for example just quickly filtering down to a known TF type and marking them as skip.

**Comment by magodo** [comment]
@jikoriko Yes, v0.4.0 doesn't include #107.

**Comment by jikoriko** [comment]
Hi @magodo, I tested with a build from the main branch and resource mapping files are now working as expected. 

I also tested with regexp and I can filter down a lot better now, for example '.ApiManagement/apis/.*/operations/' which allows me to look at just APIM Operations regardless of what api it is apart of.

The new delete key functionality also works great and speeds up the marking of items to skip great work!
